회원 정보, 테마 설정 모달

Jo-Myounghee opened the pull request:
## 작업 내용
![image](https://user-images.githubusercontent.com/48939876/140103016-8cb9deec-5a31-4684-aeb9-ad02b0d51d75.png)
![image](https://user-images.githubusercontent.com/48939876/140102823-0cc2b8ed-33b4-4b0b-82cf-71ba5a2b159e.png)
![image](https://user-images.githubusercontent.com/48939876/140102843-24989bb4-5237-41b4-8466-0cea8853cb06.png)
1. 프로필 눌렀을 때 목록 렌더링
2. 회원정보 모달
3. 테마 설정 모달
4. color, 공유 css 관련 css 리팩토링
5. 겹치는 위치 z-index 수정

## 고민한 부분
- 현재 모달창과 다른 페이지와의 z-index
- 지도 좌측 하단 네이버 로고와 second sidebar와의 관계
- 반응형을 위한 rem, % 사용
- onClick했을 때 예상과 다른 결과가 나오는 경우

## 리뷰 포인트
- 감사합니다.


## 관련된 이슈 넘버
#54 #26 #20

Jo-Myounghee committed:
Refactor: css 변수화

색상, flex css를 변수로 수정

Jo-Myounghee committed:
Refactor: color 변수 형식 통일

네이밍 변경 및 색상 HEX로 변경

Jo-Myounghee committed:
Feat: 우측 상단 프로필을 클릭하면 생성되는 계정 관리 모달

모달에는 로그아웃, 테마설정, 회원 정보 존재
토글 가능

#20

Jo-Myounghee committed:
Feat: 회원 정보 수정 모달

회원 정보(프로필 사진, 닉네임) 수정을 위한 모달창

#26

Jo-Myounghee committed:
Feat: 테마 설정 모달

테마를 설정할 수 있는 모달

#54

Jo-Myounghee committed:
Design: 모달 창과 지도 css 변경

지도, 사이드바, 모달창 순으로 z-index 증가

ddb8036631 commented on frontend/src/components/Header/Profile/Modal/ProfileModal.tsx:
```diff
@@ -0,0 +1,69 @@
+  ];
+
+  const dispatch = useDispatch();
+  const onClickProfileItem: any = (idx: number) => {
```
밑에 `onClick`에 익명 함수 호출을 해줬으니, any 타입 지정을 없애도 좋을 것 같습니다😃

ddb8036631 reviewed (approved):
잘 봤습니다👍

gofeel8 commented on frontend/src/components/Header/Profile/Modal/InnerModal/ThemeModal.tsx:
```diff
@@ -0,0 +1,109 @@
+interface UserInfoModalProps {
+  closeFn: () => void;
+  open: boolean;
+}
```
이런 타입 이제 지워주셔도 됩니다😀

gofeel8 commented on frontend/src/components/Header/Profile/Modal/InnerModal/ThemeModal.tsx:
```diff
@@ -0,0 +1,109 @@
+  ];
+
+  return (
+    <Modal open={open} closeFn={closeFn}>
```
이런 props 이제 지워주셔도 됩니다😀

gofeel8 reviewed (approved):
늦은 시간 까지 고생 많으셨습니다😴

Jo-Myounghee committed:
Fix: merge conflict fix

# Conflicts:
#	frontend/src/components/Header/index.tsx
#	frontend/src/components/Map/index.tsx
#	frontend/src/components/Modal/ModalManager/index.tsx
#	frontend/src/components/Modal/PostCreateModal/UploadImageModal/index.tsx
#	frontend/src/components/Modal/index.tsx
#	frontend/src/components/Sidebar/FirstDepth/AddGroupButton/Modal/AddGroupModal.tsx
#	frontend/src/components/Sidebar/FirstDepth/AddGroupButton/Modal/InnerModal/CreateGroupModal.tsx
#	frontend/src/components/Sidebar/FirstDepth/AddGroupButton/Modal/InnerModal/JoinGroupModal.tsx
#	frontend/src/components/Sidebar/FirstDepth/AddGroupButton/index.tsx
#	frontend/src/components/Sidebar/FirstDepth/Group/index.tsx
#	frontend/src/components/Sidebar/FirstDepth/index.tsx
#	frontend/src/components/Sidebar/SecondDepth/AddAlbum/index.tsx
#	frontend/src/components/Sidebar/SecondDepth/SettingGroup/Modal/index.tsx
#	frontend/src/components/Sidebar/SecondDepth/index.tsx
#	frontend/src/pages/Login/index.tsx
#	frontend/src/reducer/index.ts
#	frontend/src/styles/Color/index.ts
#	frontend/src/styles/StyledComponents/index.ts
